### PR TITLE
some TLC for the library

### PR DESCRIPTION
**Pull request by sneak**
This does a lot of things.

- The package now clearly errors when a python2 install attempt is made.
- It passes pep8.
- It can be installed with `setuptools` or `pipenv`.
- It includes both a `Vagrantfile` and `Dockerfile` for running a full test suite, which includes a pep8 check.
- All the console scripts have been moved to `steem/cli.py` entrypoints.

This is about ready for a new release, I think.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,55 +5,30 @@ ENV LANG en_US.UTF-8
 ENV LC_ALL en_US.UTF-8
 
 # Stuff for building steem-python
-ARG BUILD_ROOT=/buildroot
-ARG BUILD_OUTPUT=${BUILD_ROOT}/build
+ARG BUILD_ROOT=/build
```
Comment (goldibex) [inline Dockerfile]: obviously you're just renaming a thing here so this doesn't hold up the PR, but why are we using build args to begin with?

File: Pipfile [hunk 2]
```diff
@@ -1,40 +1,36 @@
...
-scrypt = "*"
-pycrypto = "*"
-requests = "*"
-urllib3 = "*"
 certifi = "*"
-ujson = "*"
-w3lib = "*"
-maya = "*"
-toolz = "*"
+ecdsa = "*"
```
Comment (goldibex) [inline Pipfile]: removing dependencies: 👍

File: steembase/memo.py [hunk 3]
```diff
@@ -144,7 +144,7 @@ def decode_memo(priv, message):
     message = aes.decrypt(unhexlify(bytes(message, 'ascii')))
     try:
         return _unpad(message.decode('utf8'), 16)
-    except:
+    except:  # noqa FIXME(sneak)
```
Comment (goldibex) [inline steembase/memo.py]: oy

Comment (sneak) [reply steembase/memo.py]: it was there when i got here. just making it pass linting so that nobody can *add* shit like this.

**Comment by goldibex** [comment]
@roadscape please review

File: steembase/memo.py (hunk 3 above)
Comment (goldibex) [reply steembase/memo.py]: oh, i know, you are the bearer of the news, not its author

File: Dockerfile (hunk 1 above)
Comment (sneak) [reply Dockerfile]: i found it like that; it could probably be removed. docker is only used presently for running the tests.

**Pull request merged**